milestones/data.py

```diff
@@ -47,17 +47,132 @@ def _get_milestone_relationship_type(relationship):
         )
 
 
+def _activate_record(record):
+    """
+    Enables database records by setting the 'active' attribute to True
+    The queries in this module filter out inactive records as part of their criteria
+    This effectively allows us to soft-delete records so they are not lost forever
+    """
+    record.active = True
+    record.save()
+
+
 def _inactivate_record(record):
     """
     Disables database records by setting the 'active' attribute to False
     The queries in this module filter out inactive records as part of their criteria
     This effectively allows us to soft-delete records so they are not lost forever
     """
-
     record.active = False
     record.save()
 
 
+def _activate_milestone(milestone):
+    """
+    Activates an inactivated (soft-deleted) milestone as well as any inactive relationships
+    """
+    [_activate_course_milestone_relationship(record) for record
+     in internal.CourseMilestone.objects.filter(milestone_id=milestone.id, active=False)]
+
+    [_activate_course_content_milestone_relationship(record) for record
+     in internal.CourseContentMilestone.objects.filter(milestone_id=milestone.id, active=False)]
+
+    [_activate_user_milestone_relationship(record) for record
+     in internal.UserMilestone.objects.filter(milestone_id=milestone.id, active=False)]
+
+    [_activate_record(record) for record
+     in internal.Milestone.objects.filter(id=milestone.id, active=False)]
```

Comment:
More of a follow-on suggestion than an insistence, but this type of multi-step DB manipulations to support a business level operation - to me - screams out the need for some DB-level transactions. I've seen some wrappers around functions in edx-platform that seem to hint at some transactionality, so perhaps there's some prior art we can replicate. @ormsbee?